pkg/kds/util/meta.go

```diff
@@ -34,21 +36,57 @@ func WithLabel(key, value string) CloneResourceMetaOpt {
 	}
 }
 
+// PopulateNamespaceLabelFromNameExtension on Kubernetes zones adds 'k8s.kuma.io/namespace' label to the resources
+// before syncing them to Global.
+//
+// In 2.7.x method 'GetMeta().GetLabels()' on Kubernetes returned a label map with 'k8s.kuma.io/namespace' added
+// dynamically. This behavior was changed in 2.9.x by https://github.com/kumahq/kuma/pull/11020, the namespace label is now
+// supposed to be set in ComputeLabels function. But this functions is called only on Create/Update of the resources.
+// This means policies that were created on 2.7.x won't have 'k8s.kuma.io/namespace' label when synced to Global.
+// Even though the lack of namespace labels affects only how resource looks in GUI on Global it's still worth setting it.
```

Comment:
There should be a cleanup task for this right?

Reply:
Added one https://github.com/kumahq/kuma/issues/11797, but it's not a straightforward cleanup, because even on 2.10 user can have policies created in 2.7